### PR TITLE
Adding fixes for signedness warnings in liblangutil

### PR DESCRIPTION
**Pull request by mijovic**

### CODE REVIEW

File: liblangutil/SourceReferenceFormatterHuman.cpp [hunk 1]
```diff
@@ -96,9 +96,9 @@ void SourceReferenceFormatterHuman::printSourceLocation(SourceReference const& _
 
 		// line 2:
 		frameColored() << line << " |";
-		m_stream << ' ' << _ref.text.substr(0, _ref.startColumn);
-		highlightColored() << _ref.text.substr(_ref.startColumn, locationLength);
-		m_stream << _ref.text.substr(_ref.endColumn) << '\n';
+		m_stream << ' ' << _ref.text.substr(0, static_cast<size_t>(_ref.startColumn));
+		highlightColored() << _ref.text.substr(static_cast<size_t>(_ref.startColumn), locationLength);
```
Comment (Marenz) [inline liblangutil/SourceReferenceFormatterHuman.cpp]: I think the failing of the tests must somehow be related to these changes

File: liblangutil/SourceReferenceExtractor.cpp [hunk 2]
```diff
@@ -58,28 +58,37 @@ SourceReference SourceReferenceExtractor::extract(SourceLocation const* _locatio
...
 		locationLength = 75;
 	}
 
 	if (line.length() > 150)
 	{
-		int const len = line.length();
-		line = line.substr(max(0, start.column - 35), min(start.column, 35) + min(locationLength + 35, len - start.column));
-		if (start.column + locationLength + 35 < len)
+		line = line.substr(
+			max(0ul, static_cast<size_t>(start.column) - 35ul),
```
Comment (a3d4) [inline liblangutil/SourceReferenceExtractor.cpp]: `max(0ul, ....` looks suspicious :).

Comment (hrkrshnn) [inline liblangutil/SourceReferenceExtractor.cpp]:
```suggestion
			static_cast<size_t>(max(0, start.column - 35)),
```

File: liblangutil/SourceReferenceFormatterHuman.cpp (hunk 1 above)
Comment (hrkrshnn) [reply liblangutil/SourceReferenceFormatterHuman.cpp]: This looks fine to me.

File: liblangutil/SourceReferenceExtractor.cpp (hunk 2 above)
Comment (hrkrshnn) [reply liblangutil/SourceReferenceExtractor.cpp]: I applied this. Let's see if that fixes the problem.

File: liblangutil/SourceReferenceExtractor.cpp [hunk 3]
```diff
@@ -58,28 +58,37 @@ SourceReference SourceReferenceExtractor::extract(SourceLocation const* _locatio
...
-		line = line.substr(max(0, start.column - 35), min(start.column, 35) + min(locationLength + 35, len - start.column));
-		if (start.column + locationLength + 35 < len)
+		line = line.substr(
+			static_cast<size_t>(max(0, start.column - 35)),
+			min(static_cast<size_t>(start.column), 35ul) + min(
+					static_cast<size_t>(locationLength) + 35ul,
+					static_cast<size_t>(static_cast<int>(line.length()) - start.column)
+			)
+		);
+		if (start.column + locationLength + 35 < static_cast<int>(line.length()))
```
Comment (hrkrshnn) [inline liblangutil/SourceReferenceExtractor.cpp]:
```suggestion
    int const len = static_cast<int>(line.length()); 
		line = line.substr(
			static_cast<size_t>(max(0, start.column - 35)),
			min(static_cast<size_t>(start.column), 35ul) + min(
					static_cast<size_t>(locationLength) + 35ul,
					static_cast<size_t>(static_cast<int>(line.length()) - start.column)
			)
		);
		if (start.column + locationLength + 35 < len)
```

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `d6fa940ab77bf46e8b5a7b01a8da60f48f8e7b83`:
```
Error: Trailing whitespace found:
 liblangutil/SourceReferenceExtractor.cpp:77: int const len = static_cast<int>(line.length()); 

```
Please check that your changes are working as intended.

File: liblangutil/SourceReferenceExtractor.cpp [hunk 4]
```diff
@@ -58,28 +58,38 @@ SourceReference SourceReferenceExtractor::extract(SourceLocation const* _locatio
...
 		line = line.substr(0, lhs) + " ... " + line.substr(rhs);
 		end.column = start.column + 75;
 		locationLength = 75;
 	}
 
 	if (line.length() > 150)
 	{
-		int const len = line.length();
-		line = line.substr(max(0, start.column - 35), min(start.column, 35) + min(locationLength + 35, len - start.column));
+    int const len = static_cast<int>(line.length());
```
Comment (stackenbotten) [inline liblangutil/SourceReferenceExtractor.cpp:77]: Coding style error

Comment (hrkrshnn) [inline liblangutil/SourceReferenceExtractor.cpp]:
```suggestion
	int const len = static_cast<int>(line.length()); 
```

File: liblangutil/SourceReferenceExtractor.cpp [hunk 5]
```diff
@@ -58,28 +58,38 @@ SourceReference SourceReferenceExtractor::extract(SourceLocation const* _locatio
...
 		line = line.substr(0, lhs) + " ... " + line.substr(rhs);
 		end.column = start.column + 75;
 		locationLength = 75;
 	}
 
 	if (line.length() > 150)
 	{
-		int const len = line.length();
-		line = line.substr(max(0, start.column - 35), min(start.column, 35) + min(locationLength + 35, len - start.column));
+	int const len = static_cast<int>(line.length());
```
Comment (stackenbotten) [inline liblangutil/SourceReferenceExtractor.cpp:77]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `3bbe2ae94308292a76f5f75c87c3767142eb2713`:
```
Error: Trailing whitespace found:
 liblangutil/SourceReferenceExtractor.cpp:77: int const len = static_cast<int>(line.length()); 

```
Please check that your changes are working as intended.

**Review by hrkrshnn — DISMISSED** [review]
Looks good to me. However, one more review would be great.

File: liblangutil/SourceReferenceExtractor.cpp [hunk 6]
```diff
@@ -58,28 +58,38 @@ SourceReference SourceReferenceExtractor::extract(SourceLocation const* _locatio
...
 	if (line.length() > 150)
 	{
-		int const len = line.length();
-		line = line.substr(max(0, start.column - 35), min(start.column, 35) + min(locationLength + 35, len - start.column));
+		int const len = static_cast<int>(line.length());
+		line = line.substr(
+			static_cast<size_t>(max(0, start.column - 35)),
+			min(static_cast<size_t>(start.column), 35ul) + min(
+					static_cast<size_t>(locationLength) + 35ul,
+					static_cast<size_t>(static_cast<int>(line.length()) - start.column)
```
Comment (hrkrshnn) [inline liblangutil/SourceReferenceExtractor.cpp]: The appveyor test fails here.

Comment (mijovic) [reply liblangutil/SourceReferenceExtractor.cpp]: I changed that now, let's see if it fixes

File: liblangutil/SemVerHandler.cpp [hunk 7]
```diff
@@ -37,7 +37,7 @@ SemVerVersion::SemVerVersion(string const& _versionString)
 	{
 		unsigned v = 0;
 		for (; i != end && '0' <= *i && *i <= '9'; ++i)
-			v = v * 10 + (*i - '0');
+			v = v * 10 + unsigned(*i - '0');
```
Comment (chriseth) [inline liblangutil/SemVerHandler.cpp:40]: Does this work for char both being signed and unsigned?

File: liblangutil/SemVerHandler.cpp [hunk 8]
```diff
@@ -245,14 +245,14 @@ unsigned SemVerMatchExpressionParser::parseVersionPart()
 		return 0;
 	else if ('1' <= c && c <= '9')
 	{
-		unsigned v = c - '0';
+		auto v = unsigned(c - '0');
```
Comment (chriseth) [inline liblangutil/SemVerHandler.cpp]: Why not `unsigned v(c - '0')`?

File: liblangutil/SemVerHandler.cpp (hunk 7 above)
Comment (mijovic) [reply liblangutil/SemVerHandler.cpp:40]: Changed to named cast instead

**Review by chriseth — APPROVED** [review]

**Pull request merged**